numba/array.py

```diff
@@ -0,0 +1,177 @@
+from numba.tracing import trace
+import numpy
+import sys
+
+def _o2s(dtype, shape, order):
+    # convert order parameter to strides
+
+    if dtype is None or shape is None or order is None:
+        return None
+
+    if order == 'F':
+        shape = list(shape)
+        shape.reverse()
+    strides = []
+    itemsize = dtype.itemsize
+    for i in range(len(shape), 0, -1):
+        strides.append(itemsize)
+        itemsize *= shape[i - 1]
+    if order in ('C', None):
+        strides.reverse()
+    return tuple(strides)
+
+def _s2o(dtype, shape, strides):
+    # convert strides parameter to order
+    # Note: strides must correspond to contiguous data layout
+
+    if strides is None or strides[-1] == dtype.itemsize:
+        order = 'C'
+    elif strides[0] == dtype.itemsize:
+        order = 'F'
+    else:
+        raise Error('strides do not correspond to contiguous data layout')
+    s2 = _o2s(dtype, shape, order)
+    if strides != s2:
+        raise Error('strides do not correspond to contiguous data layout')
+    return order
+
+class Array:
+    """An array type that supports host and GPU storage."""
+
+    def __init__(self, obj=None, copy=True,
+                 shape=None, dtype=None, order=None, where='host'):
+        """Construct an Array in the memory space defined by 'where'."""
+
```

Comment:
Indeed, good point. (In short, I was trying to emulate two overloaded constructors: one taking an array-like object as argument - with an optional 'copy' flag, the other with explicit dtype/shape/order attributes.)